File: backend/src/main/java/top/vuhe/common/util/HttpContextUtils.java
package top.vuhe.common.util;

import org.springframework.web.context.request.RequestContextHolder;
import org.springframework.web.context.request.ServletRequestAttributes;

import javax.servlet.http.HttpServletRequest;
import java.util.Objects;

/**
 * Http 上下文
 *
 * @author zhuhe
 */
public final class HttpContextUtils {
    private HttpContextUtils() {
        throw new UnsupportedOperationException("Can't instantiation!");
    }

    public static HttpServletRequest getHttpServletRequest() {
        return ((ServletRequestAttributes)
                Objects.requireNonNull(RequestContextHolder.getRequestAttributes())).getRequest();
    }

    public static String getDomain() {
        HttpServletRequest request = getHttpServletRequest();
        StringBuffer url = request.getRequestURL();
        return url.delete(url.length() - request.getRequestURI().length(), url.length()).toString();
    }

    public static String getOrigin() {
        HttpServletRequest request = getHttpServletRequest();
        return request.getHeader("Origin");
    }
}

File: backend-kt/src/main/kotlin/top/vuhe/sw/common/util/HttpContextUtils.kt
package top.vuhe.sw.common.util

import org.springframework.web.context.request.RequestContextHolder
import org.springframework.web.context.request.ServletRequestAttributes
import java.util.*
import javax.servlet.http.HttpServletRequest

fun getHttpServletRequest(): HttpServletRequest {
    return (Objects.requireNonNull(RequestContextHolder.getRequestAttributes()) as ServletRequestAttributes).request
}

fun getDomain(): String {
    val request = getHttpServletRequest()
    val url = request.requestURL
    return url.delete(url.length - request.requestURI.length, url.length).toString()
}

fun getOrigin(): String? {
    val request = getHttpServletRequest()
    return request.getHeader("Origin")
}